locator/abstract_replication_strategy.cc

```diff
@@ -178,6 +180,14 @@ long abstract_replication_strategy::parse_replication_factor(sstring rf)
     }
 }
 
+void abstract_replication_strategy::validate_options(const gms::feature_service&, const locator::topology&) const {
+    if (_uses_tablets && !is_per_table()) {
+        std::ostringstream ss;
+        ss << _my_type << " strategy is not supported with tablets";
+        throw exceptions::configuration_exception(ss.str());
```

Comment:
Would `format("{} strategy is not supported with tablets")` work instead of ss?

Reply:
I tried it and it doesn't. I know this code is ugly, but since there's a mapping of strategy types to string and a streaming operator defined, I used what was there already instead of repeating the code in a nicer form.

Reply:
Cc @tchaikov . It looks like replication_strategy_type needs its formatter

Reply:
FWIW: there is another type to string mapping `db::replication_strategy_restriction_t::map` defined in `db/config.cc`, but since we had another mapping defined in `locator/abstract_replication_strategy.cc` file, I just used it. Both mappings work the same way, but return a bit different strings. If they were to be merged together, it could lead to some tests failure.

Reply:
> Cc @tchaikov . It looks like replication_strategy_type needs its formatter

lemme create one.

Reply:
> Cc @tchaikov . It looks like replication_strategy_type needs its formatter

created #20248

Reply:
but in future, we can always use something like
```c++
fmt::format("{} strategy is not supported with tablets", fmt::streamed(meow_which_only_havs_left_shift));
```
see https://fmt.dev/11.0/api/#stdostream-support

but yeah, i agree that a proper `fmt::formatter<meow>` is better.